mantidimaging/core/io/instrument_log.py

```diff
@@ -132,3 +184,68 @@ def counts(self) -> Counts:
         counts_after = np.array(self.get_column(LogColumn.COUNTS_AFTER))
 
         return Counts(counts_after - counts_before)
+
+
+class ShutterCount:
+    """
+    Represents a shutter count log.
+
+    Attributes:
+        parsers (ClassVar[list[type[InstrumentShutterCountParser]]]): List of registered parsers.
+        parser (type[InstrumentShutterCountParser]): The parser used to parse the log.
+        data (ShutterCountType): The parsed data from the log.
+        length (int): The length of the log data.
+
+    Methods:
+        __init__(self, lines: list[str], source_file: Path): Initializes a new instance of the ShutterCount class.
+        _find_parser(self) -> None: Finds the appropriate parser for the log.
+        parse(self) -> None: Parses the log using the selected parser.
+        register_parser(cls, parser: type[InstrumentShutterCountParser]) -> None: Registers a parser for the log.
+        get_column(self, key: ShutterCountColumn) -> list[float | int]: Returns the specified column from the log data.
+        pulse_per_shutter_range_numbers(self) -> np.array: Returns an array of pulse per shutter range numbers.
+        has_Pulse(self) -> bool: Checks if the log contains the 'PULSE' column.
+        raise_if_counts_missing(self): Raises an exception if the counts are missing in the log.
+    """
+
+    parsers: ClassVar[list[type[InstrumentShutterCountParser]]] = []
+    parser: type[InstrumentShutterCountParser]
+    data: ShutterCountType
+    length: int
+
+    def __init__(self, lines: list[str], source_file: Path):
+        self.lines = lines
+        self.source_file = source_file
+
+        self._find_parser()
+        self.parse()
+
+    def _find_parser(self) -> None:
+        for parser in self.parsers:
+            if parser.match(self.lines, self.source_file.name):
+                self.parser = parser
+                return
+        raise NoParserFound
+
+    def parse(self) -> None:
+        self.data = self.parser(self.lines).parse()
+
+        lengths = [len(val) for val in self.data.values()]
+        if len(set(lengths)) != 1:
+            raise InvalidLog(f"Mismatch in column lengths: {lengths}")
+        self.length = lengths[0]
+
+    @classmethod
+    def register_parser(cls, parser: type[InstrumentShutterCountParser]) -> None:
+        cls.parsers.append(parser)
+
+    def get_column(self, key: ShutterCountColumn) -> list[float | int]:
+        return self.data[key]
+
+    def pulse_per_shutter_range_numbers(self) -> Callable[[], np.ndarray]:
+        return lambda: np.array(self.get_column(ShutterCountColumn.SHUTTER_COUNT), dtype=np.uint32)
```

Comment:
Does this need to return a lambda?

Reply:
Technically a lambda is not needed so I'll replace with:
```Python
        column_data = self.get_column(ShutterCountColumn.SHUTTER_COUNT)
        return np.array(column_data, dtype=np.uint32)
```
The lambda here is used to create an `np.array()` function from the result of `self.get_column(ShutterCountColumn.SHUTTER_COUNT)` which was just me condensing things into a concise inline definition of the function as I thought it would be a nice place to use a lambda in replacement of two lines. but probably more readable as two lines, which is usually the tradeoff with lambdas.